starlite/contrib/sqlalchemy/repository.py

```diff
@@ -0,0 +1,270 @@
+    except IntegrityError as exc:
+        raise ConflictError from exc
+    except SQLAlchemyError as exc:
+        raise RepositoryError(f"An exception occurred: {exc}") from exc
+
+
+class SQLAlchemyRepository(AbstractRepository[ModelT], Generic[ModelT]):
+    """SQLAlchemy based implementation of the repository interface."""
+
+    def __init__(self, *, session: AsyncSession, select_: Select[tuple[ModelT]] | None = None, **kwargs: Any) -> None:
+        """Repository pattern for SQLAlchemy models.
+
+        Args:
+            session: Session managing the unit-of-work for the operation.
+            select_: To facilitate customization of the underlying select query.
+        """
+        super().__init__(**kwargs)
+        self.session = session
+        self._select = select(self.model_type) if select_ is None else select_
+
+    async def add(self, data: ModelT) -> ModelT:
+        """Add `data` to the collection.
+
+        Args:
+            data: Instance to be added to the collection.
+
+        Returns:
+            The added instance.
+        """
+        with wrap_sqlalchemy_exception():
+            instance = await self._attach_to_session(data)
+            await self.session.flush()
+            await self.session.refresh(instance)
+            self.session.expunge(instance)
+            return instance
+
+    async def delete(self, id_: Any) -> ModelT:
+        """Delete instance identified by `id_`.
+
+        Args:
+            id_: Identifier of instance to be deleted.
+
+        Returns:
+            The deleted instance.
+
+        Raises:
+            RepositoryNotFoundException: If no instance found identified by `id_`.
+        """
+        with wrap_sqlalchemy_exception():
+            instance = await self.get(id_)
+            await self.session.delete(instance)
+            await self.session.flush()
+            self.session.expunge(instance)
+            return instance
+
+    async def get(self, id_: Any) -> ModelT:
+        """Get instance identified by `id_`.
+
+        Args:
+            id_: Identifier of the instance to be retrieved.
+
+        Returns:
+            The retrieved instance.
+
+        Raises:
+            RepositoryNotFoundException: If no instance found identified by `id_`.
+        """
+        with wrap_sqlalchemy_exception():
+            self._filter_select_by_kwargs(**{self.id_attribute: id_})
+            instance = (await self._execute()).scalar_one_or_none()
+            instance = self.check_not_found(instance)
+            self.session.expunge(instance)
+            return instance
+
+    async def list(self, *filters: FilterTypes, **kwargs: Any) -> list[ModelT]:
+        """Get a list of instances, optionally filtered.
+
+        Args:
+            *filters: Types for specific filtering operations.
+            **kwargs: Instance attribute value filters.
+
+        Returns:
+            The list of instances, after filtering applied.
+        """
+        for filter_ in filters:
```

Comment:
You really love that forward underscore 😅

Reply:
You might need to write a flake8 plugin :stuck_out_tongue_winking_eye: